homeassistant/components/modbus/bit_switch.py

```diff
@@ -0,0 +1,201 @@
+"""Support for Modbus switches."""
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from datetime import timedelta
+import logging
+from typing import Any
+
+from pymodbus.exceptions import ConnectionException, ModbusException
+from pymodbus.pdu import ExceptionResponse
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.const import (
+    CONF_ADDRESS,
+    CONF_NAME,
+    CONF_SCAN_INTERVAL,
+    CONF_SLAVE,
+    STATE_ON,
+)
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.helpers.restore_state import RestoreEntity
+from homeassistant.helpers.typing import DiscoveryInfoType, HomeAssistantType
+
+from .bit_sensor import ModbusReadCache
+from .const import (
+    CALL_TYPE_REGISTER_HOLDING,
+    CALL_TYPE_REGISTER_INPUT,
+    CONF_BIT_SWITCHES,
+    CONF_COMMAND_BIT_NUMBER,
+    CONF_INPUT_TYPE,
+    CONF_STATUS_BIT_NUMBER,
+    CONF_VERIFY_REGISTER,
+    CONF_VERIFY_STATE,
+    MODBUS_DOMAIN,
+)
+from .modbus import ModbusHub
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_bit_swithes(
+    hass: HomeAssistantType,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> [SwitchEntity]:
+    """Modbus Bit Switches setup."""
+    switches = []
+
+    if discovery_info is None:
```

Comment:
This function should not be called if discovery_info is None (you are the only one calling it).